torch/csrc/autograd/functions/tensor.cpp

```diff
@@ -159,12 +159,18 @@ auto CopySlices::apply(const variable_list& inputs) -> variable_list {
   // it for the backward of res. We might be able to avoid the clone() if
   // grad_slice is volatile.
   auto res = (*fn)({ grad_slice.clone() });
-  grad_slice.copy_(res[0]);
 
   variable_list grad_inputs(next_functions.size());
-  grad_inputs[0] = result;
-  for (size_t i = 1; i < res.size(); i++) {
-    grad_inputs[i] = std::move(res[i]);
+  for (size_t i = 0; i < res.size(); i++) {
+    if (should_compute_output(i)) {
+      TORCH_ASSERT(res[i].defined());
```

Comment:
I think this assertion is too strict, mostly because `should_compute_output` is really an approximation, that might sometimes have false positives. Right now it's really a check of `is_executable` of `i`th next function. However, it is possible that a function is executable, but its `i`th output was non-differentiable (e.g. it was a max index), and it was used as a target for the loss. This loss will check that its input didn't require grad at runtime, and its backward won't return grad for it.